Source/Renderer/ShaderProgram.js

```diff
@@ -27,7 +29,51 @@ define([
     /**
      * @private
      */
+
+    function extractUniforms(shaderText) {
+        // TODO: regex doesn't work for inline structs
+        var uniformNames = [];
+        var uniformLines = shaderText.match(/uniform[\w\s]*(?=[=\[;])/g);
+        if (uniformLines) {
+            for (var i = 0; i < uniformLines.length; i++) {
```

Comment:
Yes, it still matters.  The length property will be accessed each iteration and not all browsers optimize the case where it doesn't change.